edn/src/types.rs

```diff
@@ -214,9 +318,9 @@ impl Value {
     }
 }
 
-impl From<f64> for Value {
-    fn from(src: f64) -> Value {
-        Value::Float(OrderedFloat::from(src))
+impl ValueAndSpan {
+    pub fn reduce(self) -> Value {
```

Comment:
This is a reduction, but Rust tends to name general reductions `fold` and special cases like this `into_inner`, to be clear that you're consuming `self`.